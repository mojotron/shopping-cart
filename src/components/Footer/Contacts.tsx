import { CONTACT_CONFIG } from '../../constants/contacts';
import Heading from '../../ui/Heading/Heading';

const Contacts = () => {
  return (
    <div className="space-y-2">
      <Heading level={5}>Contacts</Heading>
      <ul className="space-y-3">
        {CONTACT_CONFIG.map((contact) => (
          <p key={contact.label} className="flex gap-2 items-center">
            <span className="text-green-600">{contact.icon}</span>{' '}
            <span>{contact.text}</span>
          </p>
        ))}
      </ul>
    </div>
  );
};

export default Contacts;
