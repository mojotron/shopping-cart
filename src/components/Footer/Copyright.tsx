import React from 'react';

const Copyright = () => {
  return (
    <p className="absolute bottom-5 font-bold space-x-2 text-gray-600">
      <a
        className="hover:text-orange-400"
        href="https://github.com/mojotron/shopping-cart"
        target="_blank"
        rel="noreferrer noopener"
      >
        @Mojotron
      </a>
      <span>{new Date().getFullYear()}</span>
    </p>
  );
};

export default Copyright;
