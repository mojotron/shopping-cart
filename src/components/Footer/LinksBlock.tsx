import Heading from '../../ui/Heading/Heading';

type LinkType = {
  label: string;
  path: string;
};

type PropsType = {
  label: string;
  links: LinkType[];
};

const LinksBlock = ({ label, links }: PropsType) => {
  return (
    <div>
      <Heading level={5}>{label}</Heading>
      <ul className="space-y-2">
        {links.map((link) => (
          <li
            className="cursor-pointer hover:text-green-600 text-sm text-gray-800"
            key={link.label}
          >
            <a href={link.path}>{link.label}</a>
          </li>
        ))}
      </ul>
    </div>
  );
};

export default LinksBlock;
