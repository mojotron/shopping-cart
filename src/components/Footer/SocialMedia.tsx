import {
  FaInstagram as IconIns,
  FaFacebook as IconFb,
  FaXTwitter as IconX,
} from 'react-icons/fa6';
import Heading from '../../ui/Heading/Heading';

const socialMediaConfig = [
  { label: 'instagram', icon: <IconIns size={22} /> },
  { label: 'facebook', icon: <IconFb size={22} /> },
  { label: 'x', icon: <IconX size={22} /> },
];

const SocialMedia = () => {
  return (
    <div className="space-y-2">
      <Heading level={5}>Follow us</Heading>
      <ul className="flex gap-3">
        {socialMediaConfig.map((media) => (
          <li className="cursor-pointer" key={media.label}>
            <span className="text-green-600 hover:text-orange-400">
              {media.icon}
            </span>
          </li>
        ))}
      </ul>
    </div>
  );
};

export default SocialMedia;
