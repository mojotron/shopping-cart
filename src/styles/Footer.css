.Footer {
  text-align: center;
  padding: 2rem 0 5rem;
  font-size: 1.5rem;
}
