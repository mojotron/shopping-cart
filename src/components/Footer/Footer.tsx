import Logo from '../Logo/Logo';
import Contacts from './Contacts';
import Copyright from './Copyright';
import LinksBlock from './LinksBlock';
import SocialMedia from './SocialMedia';
import {
  COMPANY_LINKS_CONFIG,
  JOIN_LINKS_CONFIG,
} from '../../constants/footerLinks';

const Footer = () => {
  return (
    <footer className="relative mt-16 flex flex-col lg:flex-row items-center gap-12 bg-gray-200 p-16 pb-24">
      <section className="flex flex-col gap-2">
        <Logo />
        <p className="">
          Eat healthy form local, organic source. Order and we will deliver in
          one hour.
        </p>

        <SocialMedia />
      </section>

      <section className="flex flex-col gap-6 sm:flex-row">
        <LinksBlock label="Company" links={COMPANY_LINKS_CONFIG} />

        <LinksBlock label="Join Our Team" links={JOIN_LINKS_CONFIG} />

        <Contacts />
      </section>
      <Copyright />
    </footer>
  );
};

export default Footer;
