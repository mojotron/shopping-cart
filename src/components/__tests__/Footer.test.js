import { render, screen } from "@testing-library/react";
import Footer from "../Footer";

describe("Footer component", () => {
  test("render footer snapshot", () => {
    const { container } = render(<Footer />);
    expect(container).toMatchSnapshot();
  });

  test("should render heading", () => {
    render(<Footer />);
    expect(screen.getByRole("heading").textContent).toMatch(
      /^created by @mojotron$/i
    );
  });

  test("should render link", () => {
    render(<Footer />);
    expect(screen.getByRole("link")).toBeInTheDocument();
  });

  test("link should have rel attribute", () => {
    render(<Footer />);
    expect(screen.getByRole("link").rel).toBe("noopener noreferrer");
  });
});
