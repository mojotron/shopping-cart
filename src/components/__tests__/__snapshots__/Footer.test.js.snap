// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer component render footer snapshot 1`] = `
<div>
  <footer
    class="Footer"
  >
    <h3
      class="Footer__info"
    >
      created by
       
      <a
        class="Footer__info__link"
        href="https://github.com/mojotron/memory-card"
        rel="noopener noreferrer"
        target="_blank"
      >
        @mojotron
      </a>
    </h3>
  </footer>
</div>
`;
